File: backend/app/schemas/artifact.py
from typing import Any, Dict, Optional
from datetime import datetime
from pydantic import BaseModel

class ArtifactResponse(BaseModel):
    id: str
    transformation_id: str
    type: str
    status: str
    title: str
    content: Optional[Dict[str, Any]] = None
    file_path: Optional[str] = None
    mime_type: Optional[str] = None
    size: int = 0
    validation_status: Optional[Dict[str, Any]] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ArtifactRegenerateRequest(BaseModel):
    custom_instructions: Optional[str] = None

File: frontend/src/types/artifact.ts
export type ArtifactType = 
  | 'executive_brief'
  | 'advisory'
  | 'social'
  | 'presentation'
  | 'infographic'
  | 'video_script'
  | 'audio'
  | 'video'
  | 'srt';

export interface ArtifactValidation {
  status: 'passed' | 'warning' | 'failed';
  warnings?: string[];
  issues?: string[];
  checks?: Record<string, boolean>;
}

export interface Artifact {
  id: string;
  transformation_id: string;
  type: ArtifactType;
  title: string;
  status: 'pending' | 'generating' | 'generated' | 'failed' | 'regenerated';
  content?: Record<string, any> | null;
  file_path?: string | null;
  mime_type?: string | null;
  size: number;
  validation_status?: ArtifactValidation | null;
  created_at: string;
  updated_at: string;
}
